### ISSUE TITLE
Active TCP Flow are not kept until ends

### ISSUE THREAD

**Issue by Alestor** [ntop/ntopng#1964]
Hello ntop Team,

I've seen in the last couple of dev versions (currently on v.3.7.180825) an unusual behavior on some long term flows in the Active Flow Overview. An active flow gets pruned after a random period of time and immediately  recreated with inaccurate Client / Server and Applications Information, because the beginning of the flow has not been seen. This  happens on TCP and ESP (IPSEC) streams, but not on UDP flows. 
My ntopng installation collects flows from a Port-Mirror. There are barley packet drops.

Received Traffic | 447.62 GB [447,348,017 Pkts] | Dropped Packets | 229 Pkts

ntopng --version
v.3.7.180825    [Enterprise/Professional build]
GIT rev:        dev:86a02ce29ad571c23d23c4e9be5eb3beeb2d6427:20180825
Pro rev:        r1853
Built on:       Debian GNU/Linux 8.2 (jessie)

Startup Line | ntopng  --user "SomeUserAccount" --data-dir "/path/to/data/dir" --interface "eth1" --dns-mode  "0" --local-networks  "<A bunch of RFC 1918 Networks>"  --dump-flows "[hidden]" --sticky-hosts "local" --dump-hosts "all"  --capture-direction "1"
-- | --

**Comment by simonemainardi** [comment]
This is normal behavior, not in ntopng per se but in any flow-based network monitoring tool. Long-lived flows are pruned if they are silent for more than a configurable interval of time.


See for example the inactive timeout at:
https://www.cisco.com/en/US/docs/ios/12_3t/netflow/command/reference/nfl_a1gt_ps5207_TSD_Products_Command_Reference_Chapter.html

Typically flows are associated with an idle and an active timeout. Those timeouts are used to prevent a flow from staying in the cache an unlimited time, as well as to make sure it is timely reported to the downstream collectors visualization tool.


Note that flows are not pruned after a random period. You can configure the flow idle timeout using the ntopng preferences:
![image](https://user-images.githubusercontent.com/3397663/44899351-c961c300-ad01-11e8-85d3-d707a8ce6a91.png)

**Comment by Alestor** [comment]
Hi @simonemainardi . I'm aware of timeouts in network monitoring solution and the configuration options for this in ntopng. The Point is, that the Flow isn't in idle state at all. It happens to active flows with permanent data exchange where no FIN / RST appears on the stream. As I wrote before "An **active** flow gets pruned after a random period of time and **immediately** recreated with inaccurate Client / Server and Applications Information, because the beginning of the flow has not been seen."

For some reason my ntop is pruning active flow in the middle of running data exchange. I've tested this with a Internet Radio Stream and there are no 60 seconds idle time for sure.
